plugins/mc_text_generator.js

```diff
@@ -0,0 +1,711 @@
+(async function() {
+
+    // Global variables
+    let aboutAction
+    let textLength, textGroup, xOffset, randNum, letterGroup
+
+    // Plugin information variables
+    const id = "mc_text_generator"
+    const name = "Minecraft Text Generator"
+    const icon = "text_format"
+    const author = "SirJain"
+
+    // Links used in about dialog
+    const links = {
+        twitter: "https://twitter.com/SirJain2",
+        discord: "https://discord.gg/wM4CKTbFVN"
+    }
+
+    // Register the plugin
+    Plugin.register(id, {
+        title: name,
+        icon,
+        author,
+        description: "Generates Minecraft-styled text in cubes.",
+        about: "This plugin adds a button under the `Tools` menu that allows you to generate Minecraft-like text.\n## How to use\nTo use this plugin, go to `Tools > Generate Text`. Simply enter some text, configure your settings how you like, and press `Generate`!\n\nPlease report any bugs or suggestions you may have.",
+        tags: ["Minecraft", "Font", "Generator"],
+        version: "1.0.0",
+        min_version: "4.2.0",
+        variant: "both",
+
+        oninstall: () => showAbout(true),
+        onload() {
+            addAboutButton()
+
+            // Define the dialog
+            const generateTextDialog = new Blockbench.Dialog({
+                name: "Generate Text",
+                icon: icon,
+                buttons: ["Generate", "Cancel"],
+                form: {
+                    input: {
+                        label: "Enter Text",
+                        type: "text",
```

Comment:
My tip: Set `full_width: true` to get a bigger text input.
You could even add a custom CSS rule via Blockbench.addCSS that sets `text-transform` to `uppercase` on this input. That way users can directly see that all texts is converted to upper case letters.